### PR TITLE
Move story locking feature to `story-editor` for consistency 

### LINKED ISSUE #10340: Decoupling: Consistent code location of story locking feature
## Task Description

In the editor we moved all story locking (`postLock`) functionality to `wp-story-editor`. But there are still some references to `postLock` in `story-editor`.

In the dashboard, all story locking functionality is still in `dashboard`, not in `wp-dashboard` as one would expect.

Should we move that to `wp-dashboard`?

Or should we move everything to the "core" packages?

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

See https://github.com/GoogleForCreators/web-stories-wp/issues/10340

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes https://github.com/GoogleForCreators/web-stories-wp/issues/10340

### CODE REVIEW

File: packages/story-editor/src/components/postLock/postLock.js [hunk 1]
```diff
@@ -52,7 +43,7 @@ function PostLock() {
     dashboardLink,
     nonce: firstNonce,
     postLock: { interval: postLockInterval, showLockedDialog },
-    api: { stories, storyLocking },
+    apiCallbacks: { getStoryLockById, setStoryLockById, deleteStoryLockById },
```
Comment (swissspidy) [inline packages/story-editor/src/components/postLock/postLock.js]: Shouldn't these come from `useAPI()`?

Comment (mohdsayed) [reply packages/story-editor/src/components/postLock/postLock.js]: Oh yes, it should. Updated!

**Comment by spacedmonkey** [comment]
Little confused by this PR. Isn't the point to move references to story locking from story-editor->wp-story-editor. This pr seems to be doing the opposite.

**Comment by swissspidy** [comment]
Please read #10340 for context.

Originally, we moved story locking to `wp-story-editor` because we felt it's more of a WordPress-specific functionality. That was quite straightforward.

However, on the dashboard side of things, the story locking functionality is still within `dashboard`, which is confusing. It turns out that extracting it to `wp-dashboard` would be rather complex and just make the code harder to maintain, with little benefit.

So one idea was to move this back to `story-editor` for consistency. Not great, but at least it sets the expectation straight for anyone implementing an editor.

That said, I am still on the fence on it, which is why on the ticket I said it's a potential maybelater/wontfix for me.

Let's continue the discussion on the ticket please.

**Comment by mohdsayed** [comment]
https://github.com/GoogleForCreators/web-stories-wp/issues/10340#issuecomment-1082780180

**Pull request closed without merging**